Code/Legacy/CrySystem/Log.cpp

```diff
@@ -897,128 +980,118 @@ void CLog::LogStringToFile(const char* szString, ELogType logType, bool bAdd, [[
     if (m_pLogIncludeTime)
     {
         uint32 dwCVarState = m_pLogIncludeTime->GetIVal();
-        //      char szTemp[MAX_TEMP_LENGTH_SIZE];
 
-        if (dwCVarState == 1)              // Log_IncludeTime
+        using TimeStringType = AZStd::fixed_string<128>;
+
+        auto GetHourMinuteSeconds = []() -> TimeStringType
         {
-            char sTime[128];
+            char sTime[TimeStringType::max_size()];
```

Comment:
question: Any reason why this is an plain old array and not `TimeStringType` too? (repeated below)

Reply:
`strftime` populates the c-string.

Because `TimeStringType` is an `AZStd::fixed_string`, supplying a pointer to its string data via `.data()` would still require a call to resize_no_construct to fix up the internal size member of the fixed_string so that the string length is correct.

Reply:
i.e The code would require an update to be
```
strftime(sTime.data(), sTime.capacity(), "<%H:%M:%S> ", &today);

// Need to fix up the m_size member of the fixed_string.
// Use the char_traits<char>::length function to calculate the c-string length
// resize_no_construct for fixed_string just, updates the m_size member to the length of the string
// and sets the value at that index to '\0'
sTime.resize_no_construct(AZStd::char_traits<char>::length(sTime.data());
```

Reply:
Ah okay I see, thank you for the additional info - would it be possible to use `AZStd::array`? Not sure if that buys us much but just wanted to mention it (the core guidelines recommend using it over plain arrays now but if they're used throughout this file likely being consistent makes the most sense).

Reply:
There is nothing in the coding standard or guidelines recommended against using C-style arrays(https://github.com/o3de/sig-core/tree/main/governance).

Reply:
Sorry I meant the core guidelines - https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#slcon1-prefer-using-stl-array-or-vector-instead-of-a-c-array - I know we don't mandate or recommend it yet

Reply:
Probably nor-should we mandate it, given we're trying to write and maintain performance sensitive code that operates in near real-time. We could easily state it's preferred to use bounds-checking array wrappers in areas where it doesn't negatively impact program performance however.

Reply:
Yeah that distinction could make sense, it's just the number of bugs that are related to out of bounds access are many 🙂 so trying to give ourselves more ways to guard against that in the majority of cases (80/20 rule) seems like a reasonable tradeoff. The interface is definitely useful too (e.g. by providing a `.size()` member function etc...)